pymongo/common.py

```diff
@@ -584,10 +606,61 @@ def validate_tzinfo(dummy, value):
     'server_selector': validate_is_callable_or_none,
 }
 
-URI_VALIDATORS.update(TIMEOUT_VALIDATORS)
-VALIDATORS = URI_VALIDATORS.copy()
+# Dictionary where keys are any URI option name, and values are the
+# internally-used names of that URI option. Options with only one name
+# variant need not be included here. Options whose public and internal
+# names are the same need not be included here.
+INTERNAL_URI_OPTION_NAME_MAP = {
+    'j': 'journal',
+    'wtimeout': 'wtimeoutms',
+    'tls': 'ssl',
+    'tlsallowinvalidcertificates': 'ssl_cert_reqs',
+    'tlsallowinvalidhostnames': 'ssl_match_hostname',
+    'tlscrlfile': 'ssl_crlfile',
+    'tlscafile': 'ssl_ca_certs',
+    'tlsclientcertfile': 'ssl_certfile',
+    'tlsclientkeypassword': 'ssl_pem_passphrase',
+    'tlsclientkeyfile': 'ssl_keyfile',
+}
+
+# Map from deprecated URI option names to the updated option names.
+# Case is preserved for updated option names as they are part of user warnings.
+URI_OPTIONS_DEPRECATION_MAP = {
+    'j': 'journal',
```

Comment:
Does this change flip the priority of 'j' and 'journal'? It looks like before 'j' would override 'journal' but now it's reversed. I'm not too concerned but we should at least document all potentially breaking changes like this in the changelog.